tests/test_forward_ref.py

```diff
@@ -758,13 +764,9 @@ class Bar(BaseModel, Generic[T]):
     finally:
         del sys.modules['eval_type_backport']
 
-    assert module.Foo.model_fields['bar'].annotation == typing.Optional[module.Bar[str]]
-    assert module.Foo.model_fields['bar2'].annotation == typing.Union[int, module.Bar[float]]
```

Comment:
These assertions are not valid anymore, as the `Bar[str]` class on the annotation is no longer the same that the one we create by calling `module.Bar[str]` here (as it isn't cached anymore).